eng/scripts/Compile-Changelog.ps1

```diff
@@ -0,0 +1,711 @@
+    Install-Module -Name powershell-yaml -Force -Scope CurrentUser -AllowClobber
+}
+Import-Module powershell-yaml -ErrorAction Stop
+
+# Parse and validate YAML files
+$entries = @()
+$validSections = @("Features Added", "Breaking Changes", "Bugs Fixed", "Other Changes")
+
+foreach ($file in $yamlFiles) {
+    Write-Host "Processing: $($file.Name)" -ForegroundColor Gray
+    
+    # Validate filename format (should be numeric timestamp)
+    if ($file.BaseName -notmatch '^\d+$') {
+        Write-Warning "  Filename '$($file.Name)' doesn't follow timestamp convention (numeric only)"
+    }
+    
+    try {
+        $yamlContent = Get-Content -Path $file.FullName -Raw
+        $entry = $yamlContent | ConvertFrom-Yaml
+        
+        # Validate required fields
+        if (-not $entry.section) {
+            Write-Error "  Missing required field 'section' in $($file.Name)"
+            continue
+        }
+        
+        # Validate and normalize section (case-insensitive)
+        $matchedSection = $validSections | Where-Object { $_ -ieq $entry.section }
+        if ($matchedSection) {
+            # Use the properly cased version
+            $normalizedSection = $matchedSection
+        } else {
+            Write-Error "  Invalid section '$($entry.section)' in $($file.Name). Must be one of: $($validSections -join ', ')"
+            continue
+        }
+        
+        if (-not $entry.description) {
+            Write-Error "  Missing required field 'description' in $($file.Name)"
+            continue
+        }
+        
+        if ($entry.description.Length -lt 10) {
+            Write-Error "  Description too short in $($file.Name) (minimum 10 characters)"
+            continue
+        }
+        
+        if (-not $entry.pr -or $entry.pr -eq 0) {
+            Write-Warning "  Missing or invalid PR number in $($file.Name)"
+        }
+        elseif ($entry.pr -lt 1) {
+            Write-Error "  Invalid PR number in $($file.Name) (must be positive)"
+            continue
+        }
+        
+        # Add to entries collection with normalized section name
+        $entries += [PSCustomObject]@{
+            Section = $normalizedSection
+            Subsection = if ($entry.subsection -and $entry.subsection -ne "null") { $entry.subsection } else { $null }
+            Description = $entry.description
+            PR = $entry.pr
+            Filename = $file.Name
+        }
+        
+        Write-Host "  ✓ Valid" -ForegroundColor Green
+    }
+    catch {
+        Write-Error "  Failed to parse $($file.Name): $_"
+        continue
+    }
+}
+
+if ($entries.Count -eq 0) {
+    Write-Error "No valid changelog entries found"
+    exit 1
+}
+
+Write-Host ""
+Write-Host "Successfully validated $($entries.Count) entry/entries" -ForegroundColor Green
+Write-Host ""
+
+# Group entries by section and subsection
+$sectionOrder = @("Features Added", "Breaking Changes", "Bugs Fixed", "Other Changes")
```

Comment:
Any chance we can infer this from the schema, so if we add more sections to the schema, the order is automatically updated here?